doc/md/language-manual.md

```diff
@@ -2022,6 +2036,17 @@ The context type is a record to allow extension with further fields in future re
 
 :::
 
+Shared functions have different capabilities dependent on their qualification as `shared`, `shared query` or `shared composite query`.
+
+A `shared` function may call any `shared` or `shared query` function, but no `shared composite query` function.
```

Comment:
I heard that update calls cannot invoke composite queries. Just for curiosity: Is it a technical reason? Or a conceptual aspect?

Reply:
When an update calls a query, the query is actually executed in replicated mode for concensus on the result (state change is still discarded). 
Similary, inspect message is only called for query functions that are executed in update mode (with concensus), not ordinary fast queries.
I suspect that replicated composite queries are much harder to implement and haven't been tackled yet, and might not be.

Reply:
Thanks for the explanation.